test/evergreen.yml

```diff
@@ -694,18 +694,6 @@ tasks:
     tags: ["pull_request", "pull_request_compilers"]
     commands:
       - func: "get project"
-      - func: "compile wiredtiger"
-        vars:
-          configure_env_vars: CC=gcc-4.8 CXX=g++-4.8 ADD_CFLAGS="-ggdb -fPIC"
-      - func: "compile wiredtiger"
-        vars:
-          configure_env_vars: CC=gcc-5 CXX=g++-5 ADD_CFLAGS="-ggdb -fPIC"
-      - func: "compile wiredtiger"
-        vars:
-          configure_env_vars: CC=gcc-6 CXX=g++-6 ADD_CFLAGS="-ggdb -fPIC"
-      - func: "compile wiredtiger"
-        vars:
-          configure_env_vars: CC=gcc-6 CXX=g++-6 ADD_CFLAGS="-ggdb -fPIC" CPPFLAGS=-fvisibility=hidden
```

Comment:
I think I missed the discussion about removing these legacy gcc/clang versions. Can you provide some context?

Reply:
Monica told me that this change was approved by Alex C,  he said that the older versions of GCC/clang were quite obsolete and we do not see a strong reason to continue supporting it. Also, some of them are not available on Ubuntu 20.04.

Reply:
Forgive my memory. If that's the case I'd expect this is documented as a design decision on the jira ticket.

Reply:
@keithbostic do you have any opinion on the decision of removing testing coverage for those older gcc/clang versions?

Reply:
@lukech, I think it's reasonable to drop gcc 4 and 5; gcc 6 is 5 years old (released in April 2016), so I'd say it's borderline. I wouldn't fight hard to keep gcc 6, I don't think we're making a mistake to drop it if it's difficult to keep building with it.

If we drop those compilers, there may be additional cleanups we can make in our build systems -- for autotools, there's `build_posix/aclocal/strict.m4` that includes gcc 4 and 5 compiler support, I don't know what CMake does for various compilers.

Reply:
@keithbostic, thanks for the input. 
@alisonfel, for your context - we are planning to stop testing older gcc/clang versions starting from this PR. Hope that can simplify some CMake code.